service/matching/matchingEngine.go

```diff
@@ -853,6 +857,28 @@ func (e *matchingEngineImpl) GetTaskQueueUserData(
 	return resp, nil
 }
 
+func (e *matchingEngineImpl) ApplyTaskQueueUserDataReplicationEvent(
+	hCtx *handlerContext,
+	req *matchingservice.ApplyTaskQueueUserDataReplicationEventRequest,
+) (*matchingservice.ApplyTaskQueueUserDataReplicationEventResponse, error) {
+	namespaceID := namespace.ID(req.GetNamespaceId())
+	taskQueueName := req.GetTaskQueue()
+	taskQueue, err := newTaskQueueID(namespaceID, taskQueueName, enumspb.TASK_QUEUE_TYPE_WORKFLOW)
+	if err != nil {
+		return nil, err
+	}
+	tqMgr, err := e.getTaskQueueManager(hCtx, taskQueue, enumspb.TASK_QUEUE_KIND_NORMAL, true)
+	if err != nil {
+		return nil, err
+	}
+	err = tqMgr.UpdateUserData(hCtx, false, func(current *persistencespb.TaskQueueUserData) (*persistencespb.TaskQueueUserData, error) {
+		mergedUserData := *current
```

Comment:
just checking, is a shallow copy enough here? i.e. MergeVersioningData doesn't mutate deeper structure?